test/controllers/sessions_controller_test.rb

```diff
@@ -2,4 +2,74 @@
 
 describe SessionsController do
 
+  describe 'create' do
+    it 'can login an existing user' do
+      user = users(:grace)
+
+      # Tell OmniAuth to use this user's info when it sees
+      # an auth callback from github
+      # call test helper method mock_auth
+      OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new(mock_auth_hash(user))
```

Comment:
So your `before`s weren't working because of a bug in Mini-Test, but here is where I would instead be calling `perform_login`.

Since `before` is broken, the best way to fix is to use `let` instead.